code/datums/extensions/health/_health.dm

```diff
@@ -0,0 +1,175 @@
+ * Provided as a proc for override use.
+ */
+/datum/extension/health/proc/current_damage(damage_type = null)
+	return max_health - current_health
+
+
+/**
+ * Retrieves the extension's death state.
+ * Provided as a proc for override use.
+ */
+/datum/extension/health/proc/get_death_state()
+	return death_state
+
+
+/**
+ * Sets a new death_state.
+ * Returns `TRUE` of the new state is different from the old state, `FALSE` otherwise.
+ * Also calls `handle_death_change()` on the holder.
+ */
+/datum/extension/health/proc/set_death_state(new_death_state)
+	if (death_state == new_death_state)
+		return FALSE
+	death_state = new_death_state
+	var/atom/A = holder
+	A.handle_death_change(new_death_state)
+	return TRUE
+
+
+/**
+ * Checks the current condition of health and sets the death state accordingly.
+ * Returns `TRUE` if `death_state` was changed, `FALSE` otherwise.
+ */
+/datum/extension/health/proc/update_death_state()
+	if (current_health() == 0)
+		return set_death_state(TRUE)
+	else
+		return set_death_state(FALSE)
+
+
+/**
+ * Modifies `current_health` by the given modifier.
+ * Returns `TRUE` if `death_state` was changed, `FALSE` otherwise.
+ */
+/datum/extension/health/proc/mod_health(health_mod, damage_type = null)
+	SHOULD_CALL_PARENT(TRUE)
+
+	// Apply resistances to negative modifiers
+	if (health_mod < 0 && damage_type && LAZYISIN(damage_resistances, damage_type))
+		health_mod = health_mod * damage_resistances[damage_type]
+
+	health_mod = round(health_mod, 1)
+	if (health_mod == 0)
+		return FALSE // No health change, no further processing needed
+	current_health = Clamp(current_health + health_mod, 0, max_health())
+
+	var/atom/A = holder
+	if (health_mod > 0)
+		A.post_restore_health(health_mod, damage_type)
+	else if (health_mod < 0)
+		A.post_damage_health(abs(health_mod), damage_type)
+	A.post_health_change(health_mod, damage_type)
+
+	return update_death_state()
+
+
+/**
+ * Immediately sets health to `0` then updates `death_state`, bypassing all other checks and processes.
+ */
+/datum/extension/health/proc/kill()
+	SHOULD_CALL_PARENT(TRUE)
+	current_health = 0
+	return update_death_state()
+
+
+/**
+ * Immediately restores health to full then updates `death_state`, bypassing all other checks and processes.
+ */
+/datum/extension/health/proc/revive()
+	SHOULD_CALL_PARENT(TRUE)
+	current_health = max_health()
+	return update_death_state()
+
+
+/**
+ * Changes `max_health` to a new value, with a minimum of `1`.
+ * Returns `TRUE` if the health changes causes a death state change, `FALSE` otherwise.
+ */
+/datum/extension/health/proc/set_max_health(new_max_health)
+	SHOULD_CALL_PARENT(TRUE)
+	if (new_max_health == max_health)
+		return FALSE
+	max_health = max(round(new_max_health, 1), 1)
+	current_health = max_health()
+	return update_death_state()
+
+
+/**
+ * Clones health data from another health extension.
+ */
+/datum/extension/health/proc/clone(datum/extension/health/source)
```

Comment:
This function works differently from what you'd expect a `clone()` function to -- `copy_from()` would be a better name.

Reply:
I renamed it to `copy_health()` to match the atom proc that uses it.